### ISSUE TITLE
Dark mode for canvas

### ISSUE THREAD

**Issue by coldspark29** [xournalpp/xournalpp#1781]
**Is your feature request related to a problem? Please describe.**
When I study at night, it is very stressful for my eyes to look at a white canvas. So I turned the background black and I could study or two more hours. The problem was that you can't turn all pages black at once. You have to do it for every single page.

**Describe the solution you'd like**
I would like to have a dark mode activated by clicking a button, where you could set a dark color of your choice as background. Black or almost black colors should be inverted and the colors like green, red and blue could stay the same. This would of course require a bit of knowledge of color codes to set a threshold of which colors should be inverted. 

**Describe alternatives you've considered**
The easiest implementation would turning the white canvas black and all black strokes white.

**Comment by Fatih20** [comment]
I second this. It would be really great.

**Comment by noravanq** [comment]
Agree, this would be great. Ideally you should be able to switch between the modes easily, so you can write in dark mode but print in light mode.

**Comment by coldspark29** [comment]
Yes, in fact printing should always be in light mode :) I also thought that finding an algorithm that detects color codes intelligently will be too much of a hassle. I think two modes would be nice.

1. You invert all colors
2. You invert only black and white colors, so the other colors stay the same and don't look like airport scanners

I think this should be easy to implement!

**Comment by Fatih20** [comment]
2nd option of course. It's obvious.

**Comment by coldspark29** [comment]
I think there are some images for which inverting everything will be better. If you have dark colors on a white canvas and you turn the canvas black, it will be barely readable. So I think two modes should be available.

OR you use the second option and turn up the brightness for those colors to a degree where they are readable. So you basically block colors that are not readable on a black canvas. But well, that would be really a lot of work and needs a lot of knowledge about color codes. I don't know if anything like that even exists yet.

**Comment by coldspark29** [comment]
On the other hand we are not talking about photographs here, but about simple notes. So maybe what I am having in mind does not really have that much of an impact...

**Comment by Fatih20** [comment]
Adding on to this. I think we should also have dark color for the background. Here's an example from LibreOffice.
![image](https://user-images.githubusercontent.com/51940383/80277448-3157ef00-8719-11ea-98c1-5ccf60923fd6.png)

**Comment by fanzhuyifan** [comment]
Fyi, here is some information on "smart" inversion of colors:

https://github.com/vn971/linux-color-inversion
